### PR TITLE
Publish Reserve econ data: mint & burn

### LINKED ISSUE #4651: Expose key Reserve Contract economic data via subscription
This data can be used in Blockchain Explorers and via sites like The Graph for the community to get key information about the state of the RUN Protocol components.

From [our meeting on Feb 22](https://docs.google.com/document/d/1HkNLG8Kmm9DV6XGxvZEGVWUtHli2OfkR5ph7Q77x3QE/edit#), here is the data we want to expose from the Reserve Contract:

1. How much of each asset type
2. Governance parameters

We had initially thought the Reward Pool would be housed here, but @Chris-Hibbert wrote: "The collateral reserve contract holds funds from various sources. Its main goal is holding assets that governance decisions can direct to the AMM or other uses. It is unlikely to be the main repository for reward funds."

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
closes: #4651

## Description

Publish key economic data (minting and burning) from the Reserve contract.

### Security Considerations

publishing data that should be public.  No security concerns.

### Documentation Considerations

None.

### Testing Considerations

Tested that the values were published.

### CODE REVIEW

File: packages/run-protocol/src/reserve/assetReserve.js [hunk 1]
```diff
@@ -25,6 +25,8 @@ const nonalphanumeric = /[^A-Za-z0-9]/g;
  *
  * @property {AmountKeywordRecord} allocations
  * @property {Amount<'nat'>} shortfallBalance shortfall from liquiditation that
+ * @property {Amount<'nat'>} mintedRUN total RUN minted to date
+ * @property {Amount<'nat'>} burnedRUN total RUN burned to date
```
Comment (turadg) [inline packages/run-protocol/src/reserve/assetReserve.js]: let's use `Fee` instead of `RUN` per https://github.com/Agoric/agoric-sdk/issues/4800

and per https://github.com/Agoric/agoric-sdk/blob/fb4b156bdd74e28af8fb93f2da3f43c1041c1cdf/packages/run-protocol/src/vaultFactory/vaultManager.js#L46 `minted` should be second. 

"total" helps convey that this is "minted _ever_" not any other time window.

```suggestion
 * @property {Amount<'nat'>} totalFeeMinted total Fee minted to date
 * @property {Amount<'nat'>} totalFeeBurned total Fee burned to date
```

File: packages/run-protocol/src/reserve/assetReserve.js [hunk 2]
```diff
@@ -222,6 +224,9 @@ const start = async (zcf, privateArgs) => {
 
   const { brand: runBrand } = await E(runMint).getIssuerRecord();
 
+  let mintedRUN = AmountMath.makeEmpty(runBrand);
+  let burnedRUN = AmountMath.makeEmpty(runBrand);
```
Comment (turadg) [inline packages/run-protocol/src/reserve/assetReserve.js]: just noting the variable names should continue to match the metric names

File: packages/run-protocol/test/metrics.js [hunk 3]
```diff
@@ -102,3 +102,9 @@ export const vaultManagerMetricsTracker = async (t, publicFacet) => {
     assertFullyLiquidated,
   });
 };
+export const reserveInitialState = emptyRun => ({
+  allocations: {},
+  shortfallBalance: emptyRun,
+  burnedRUN: emptyRun,
+  mintedRUN: emptyRun,
+});
```
Comment (turadg) [inline packages/run-protocol/test/metrics.js]: this is valid only for `test-reserve.js` so please keep it there

File: packages/run-protocol/src/reserve/assetReserve.js [hunk 4]
```diff
@@ -316,6 +325,7 @@ const start = async (zcf, privateArgs) => {
       }),
     );
     zcf.reallocate(offerToSeat, collateralSeat);
+    updateMetrics();
```
Comment (turadg) [inline packages/run-protocol/src/reserve/assetReserve.js:328]: why is this needed here but not after `burnedRUN` change?

**Review by turadg — CHANGES_REQUESTED** [review]

File: packages/run-protocol/test/metrics.js (hunk 3 above)
Comment (Chris-Hibbert) [reply packages/run-protocol/test/metrics.js]: I hoisted it when the vault test needed it as well. I didn't want the vault test to have to be updated if we add more to the record.  I didn't see a better place to put it, and `metrics.js` is at least under `test`.

File: packages/run-protocol/src/reserve/assetReserve.js (hunk 1 above)
Comment (Chris-Hibbert) [reply packages/run-protocol/src/reserve/assetReserve.js]: Done

File: packages/run-protocol/src/reserve/assetReserve.js (hunk 2 above)
Comment (Chris-Hibbert) [reply packages/run-protocol/src/reserve/assetReserve.js]: done

File: packages/run-protocol/src/reserve/assetReserve.js (hunk 4 above)
Comment (Chris-Hibbert) [reply packages/run-protocol/src/reserve/assetReserve.js:328]: It is already called in `burnRUNToReduceShortfall`, because that calls `increaseLiquidationShortfall`. `addLiquiditytoAmmPool` mints, but doesn't adjust the shortfall. The shortfall is increased by other contracts that get a `shortfallReportingInvitation`.

File: packages/run-protocol/test/metrics.js (hunk 3 above)
Comment (turadg) [reply packages/run-protocol/test/metrics.js]: ah, my mistake that it's used only in `test-reserve.js`.

**Review by turadg — APPROVED** [review]
